### ISSUE TITLE
php5 openssl.so is missing

### ISSUE THREAD

**Issue by justClouds** [Microsoft/BashOnWindows#380]
Running a website on Apache gives some errors in PHP. There are several libraries in /usr/lib/php5/20121212/ but no openssl.so.

**Comment by benhillis** [comment]
Could you install the libssl-dev package and try again?

`apt-get install libssl-dev`

**Comment by justClouds** [comment]
I did (including another apt-get update/upgrade), but still no openssl.so in /usr/lib/php5/20121212/. And still the same error:
`PHP Warning:  PHP Startup: Unable to load dynamic library '/usr/lib/php5/20121212/openssl.so' - /usr/lib/php5/20121212/openssl.so: cannot open shared object file: No such file or directory in Unknown on line 0`
This happens when installing a package in ([Grav](https://getgrav.org), with the Grav package manager GPM (a php script). Works fine in Ubuntu on Azure.

**Comment by aseering** [comment]
I'm not familiar with Grav; could you post the exact commands that you're using to install it?  Also, ideally, a complete log of the output of the command that's failing?

Just to make sure -- you're comparing against an Ubuntu 14.04 instance in Azure, not something newer?

`apt-file` can't find the file `/usr/lib/php5/20121212/openssl.so` anywhere in the official Ubuntu repositories.  I don't see it on a real (bare-metal) Ubuntu server.  I'm guessing it's something that GPM is supposed to put there.  If that's the case, then the fact that it's not putting it there implies to me that there's a different error earlier in the script that's (closer to) the root cause.

**Comment by aseering** [comment]
One observation -- Bash-on-Windows ships with a somewhat minimal set of preinstalled packages.  I'm guessing that Azure's image ships with more preinstalled PHP-related packages.  Could be a missing dependency; probably what @benhillis was getting at.

If you happen to be familiar with Docker, and can reproduce this issue inside the stock Ubuntu 14.04 Docker Container, then it probably is a missing dependency and therefore a bug in some GPM package (Grav or a dependency).  (Docker's official Ubuntu image is extremely minimal, therefore very good at catching missing-dependency packaging bugs.)

**Comment by justClouds** [comment]
Thanks for the suggestions.
Unfortunately, the Ubuntu test VM is no longer running, so I am not 100% sure what version I used for the test ;o). Will try with Docker container and keep an eye on the php library directory.

I don't expect it to be a module installed by Grav as that is installed as just a bunch of files/directories inside the apache webroot (simple cp command). It is not a .deb and has no defined dependencies as such. I did install git, apache2, php5, php5-cli, php5-gd, php5-curl, libapache2-mod-php5 as they are prequisite,

As a simple test I created a minimal php script:
`<?php phpinfo(); ?>
`
Executing that script gives the same error...?!?
So I removed and reinstalled the php modules just to be sure: no improvement. It seems like an error I should just ignore for now as the php scripts seem to be running fine otherwise.
